packages/typescript-vfs/src/index.ts

```diff
@@ -371,7 +375,44 @@ export function createSystem(files: Map<string, string>): System {
     write: () => notImplemented("write"),
     writeFile: (fileName, contents) => {
       files.set(fileName, contents)
-    }
+    },
+  }
+}
+
+/**
+ * Creates a file-system backed System object which can be used in a TypeScript program, this
+ * is what provides read/write aspects of the virtual fs
+ */
+export function createFSBackedSystem(files: Map<string, string>): System {
+  const fs = require("fs")
+
+  return {
+    args: [],
+    createDirectory: () => notImplemented("createDirectory"),
+    // TODO: could make a real file tree
+    directoryExists: audit("directoryExists", directory => {
+      return Array.from(files.keys()).some(path => path.startsWith(directory)) || fs.existsSync(directory)
+    }),
+    exit: () => notImplemented("exit"),
+    fileExists: audit(
+      "fileExists",
+      fileName => files.has(fileName) || files.has(libize(fileName)) || fs.existsSync(fileName)
+    ),
+    getCurrentDirectory: () => "/",
+    getDirectories: () => [],
+    getExecutingFilePath: () => notImplemented("getExecutingFilePath"),
+    readDirectory: audit("readDirectory", directory => (directory === "/" ? Array.from(files.keys()) : [])),
+    readFile: audit(
+      "readFile",
+      fileName => files.get(fileName) || files.get(libize(fileName) || fs.readFileSync(fileName, { encoding: "utf-8" }))
+    ),
```

Comment:
I think this might be a bit too naive, as it looks like the current dir is "/" and I'm not sure it can resolve to your projects node modules, will take a look

Reply:
Perhaps the answer is to provide a FS root to work from